flocker/node/script.py

```diff
@@ -148,9 +154,18 @@ class ZFSAgentScript(object):
     a Flocker cluster.
     """
     def main(self, reactor, options, volume_service):
-        configuration = agent_config_from_file(path=options[u'agent-config'])
+        try:
+            agent_config = options[u'agent-config']
+            configuration = yaml.safe_load(agent_config.getContent())
+        except IOError:
+            raise ConfigurationError(
+                "Configuration file does not exist at '{}'.".format(
+                    agent_config.path))
```

Comment:
Same comment about exception-handling anti-pattern here.

Also, this looks section looks like an exact copy of the code from `AgentServiceFactory`.  Another reason to put it in a function, sounds like.  Please make a note of this in that issue and link to it from here as well.